addons/compat_cup_units/CfgGlasses.hpp

```diff
@@ -0,0 +1,492 @@
+        ace_resistance = 1;
+        ESS_OVERLAY;
+    };
+    class CUP_G_ESS_RGR_Facewrap_Tropical: None {
+        ace_protection = 1;
+        ace_resistance = 1;
+        ESS_OVERLAY;
+    };
+
+    // Oakleys - Dark
+    class CUP_G_Oakleys_Drk: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Black_Glasses_Dark: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Black_Glasses_Dark_Headset: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tan_Glasses_Dark: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tan_Glasses_Dark_Headset: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tropical_Glasses_Dark: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tropical_Glasses_Dark_Headset: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Winter_Glasses_Dark: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Winter_Glasses_Dark_Headset: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_RadioHeadset_Glasses_Dark: CUP_G_PMC_RadioHeadset_Glasses {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+
+    // Oakleys - Ember
+    class CUP_G_Oakleys_Embr: None {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Black_Glasses_Ember: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tan_Glasses_Ember: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Tropical_Glasses_Ember: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_Facewrap_Winter_Glasses_Ember: CUP_G_PMC_Facewrap_Black_Glasses_Dark {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_PMC_RadioHeadset_Glasses_Ember: CUP_G_PMC_RadioHeadset_Glasses {
+        ace_color[] = {1, 0, 0};
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+
+    // Shades - Dark
+    class CUP_G_Beard_Shades_Black: None {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+    class CUP_G_Beard_Shades_Blonde: CUP_G_Beard_Shades_Black {
+        ace_resistance = 1;
+        ace_tintAmount = 8;
+    };
+
+    // Shades - Clear
```

Comment:
How come no inheritance is used for the diff variants ? These shades being a good place to, for ex, which would decrease space wasting.

Reply:
Ask CUP, it is how they encoded it.

Reply:
This is only related to ACE, that's why.